plugins/woocommerce/tests/php/includes/data-stores/class-wc-order-data-store-cpt-test.php

```diff
@@ -466,4 +466,50 @@ public function test_create_draft_order_doesnt_trigger_hook() {
 
 		remove_action( 'woocommerce_new_order', $callback );
 	}
+
+	/**
+	 * @testDox Updating an order status correctly triggers the "woocommerce_new_order" action.
+	 */
+	public function test_update_order_status_correctly_triggers_new_order_hook() {
+
+		$new_count = 0;
+
+		$callback = function () use ( &$new_count ) {
+			++$new_count;
+		};
+
+		add_action( 'woocommerce_new_order', $callback );
+
+		$nonTriggeringOrderStatuses = array( 'checkout-draft', 'refunded', 'failed', 'cancelled' );
+
+		$order_data_store_cpt = new WC_Order_Data_Store_CPT();
+
+		$order = new WC_Order();
```

Comment:
I would add another test case here to create a new order, set status to `processing` immediately, and update. It should trigger `woocommerce_new_order` but I suspect it will not until my other feedback is resolved.

Reply:
Yes 👍🏻